fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadTaskScheduler.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.doris.task.AgentTaskQueue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Queue;
+
+/**
+ * Routine load task scheduler is a function which allocate task to be.
+ * Step1: get total idle task num of backends.
+ * Step2: equally divide to be
+ * Step3: submit tasks to be
+ */
+public class RoutineLoadTaskScheduler extends Daemon {
+
+    private static final Logger LOG = LogManager.getLogger(RoutineLoadTaskScheduler.class);
+
+    private RoutineLoad routineLoad = Catalog.getInstance().getRoutineLoadInstance();
+
+    @Override
+    protected void runOneCycle() {
+        // update current beIdMaps for tasks
+        routineLoad.updateBeIdTaskMaps();
+
+        // check timeout tasks
+        routineLoad.processTimeOutTasks();
+
+        // get idle be task num
+        int totalIdleTaskNum = routineLoad.getTotalIdleTaskNum();
+        int allocatedTaskNum = 0;
+        Queue<RoutineLoadTaskInfo> routineLoadTaskList = routineLoad.getNeedSchedulerRoutineLoadTasks();
+        AgentBatchTask batchTask = new AgentBatchTask();
+
+        // allocate task to be
+        while (totalIdleTaskNum > 0) {
+            RoutineLoadTaskInfo routineLoadTaskInfo = routineLoadTaskList.poll();
+            // queue is not empty
+            if (routineLoadTaskInfo != null) {
+                // when routine load task is not abandoned
+                if (routineLoad.getIdToRoutineLoadTask().get(routineLoadTaskInfo.getSignature()) != null) {
+                    long beId = routineLoad.getMinTaskBeId();
+                    RoutineLoadJob routineLoadJob = routineLoad.getJobByTaskId(routineLoadTaskInfo.getSignature());
+                    RoutineLoadTask routineLoadTask = null;
+                    if (routineLoadTaskInfo instanceof KafkaTaskInfo) {
+                        routineLoadTask = new KafkaRoutineLoadTask(routineLoadJob.getResourceInfo(),
+                                beId, routineLoadJob.getDbId(), routineLoadJob.getTableId(),
+                                0L, 0L, 0L, routineLoadJob.getColumns(),
+                                routineLoadJob.getWhere(), routineLoadJob.getColumnSeparator(),
+                                (KafkaTaskInfo) routineLoadTaskInfo,
+                                ((KafkaRoutineLoadJob) routineLoadJob).getProgress());
+                    }
+                    if (routineLoadTask != null) {
+                        routineLoadTaskInfo.setLoadStartTimeMs(System.currentTimeMillis());
+                        AgentTaskQueue.addTask(routineLoadTask);
+                        batchTask.addTask(routineLoadTask);
+                        totalIdleTaskNum--;
+                        allocatedTaskNum++;
+                        routineLoad.addNumOfConcurrentTasksByBeId(beId);
+                    }
+                } else {
+                    LOG.debug("Task {} for job already has been discarded", routineLoadTaskInfo.getSignature());
```

Comment:
The correct grammar is: 'has already been' or 'has been already', not 'already has been'